src/EditorFeatures/Core/EmbeddedLanguages/RegularExpressions/RegexEmbeddedBraceMatcher.cs

```diff
@@ -1,35 +1,37 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor;
 using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
 using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices;
 using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
 using Microsoft.CodeAnalysis.LanguageServices;
 using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.Utilities;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages.RegularExpressions
 {
-    using System.Collections.Immutable;
     using RegexToken = EmbeddedSyntaxToken<RegexKind>;
     using RegexTrivia = EmbeddedSyntaxTrivia<RegexKind>;
 
     /// <summary>
     /// Brace matching impl for embedded regex strings.
     /// </summary>
-    internal sealed class RegexEmbeddedBraceMatcher : IEmbeddedBraceMatcher
+    internal sealed class RegexEmbeddedBraceMatcher : IBraceMatcher
     {
-        private readonly RegexEmbeddedLanguage _language;
+        private readonly EmbeddedLanguageInfo _info;
 
-        public RegexEmbeddedBraceMatcher(RegexEmbeddedLanguage language)
+        public RegexEmbeddedBraceMatcher(EmbeddedLanguageInfo info)
         {
-            _language = language;
+            _info = info;
         }
 
-        public async Task<EmbeddedBraceMatchingResult?> FindBracesAsync(
+        public async Task<BraceMatchingResult?> FindBracesAsync(
```

Comment:
No need for duplicated types so the Workspace layer would knwo about these Feature/Editor concepts.